spec/meilisearch/client/keys_spec.rb

```diff
@@ -1,60 +1,143 @@
 # frozen_string_literal: true
 
 RSpec.describe 'MeiliSearch::Client - Keys' do
-  it 'gets the list of keys' do
-    response = client.keys
-    expect(response).to be_a(Hash)
-    expect(response.count).to eq(2)
-    expect(response.keys).to contain_exactly('private', 'public')
-    expect(response['private']).to be_a(String)
-    expect(response['public']).to be_a(String)
-  end
+  context 'Test the default key roles' do
+    let(:public_key) { client.keys['results'].filter { |k| k['description'].start_with? 'Default Search API Key' }.first }
+    let(:private_key) { client.keys['results'].filter { |k| k['description'].start_with? 'Default Admin API Key' }.first }
 
-  it 'fails to get settings if public key used' do
-    public_key = client.keys['public']
-    new_client = MeiliSearch::Client.new(URL, public_key)
-    expect do
-      new_client.index(random_uid).settings
-    end.to raise_meilisearch_api_error_with(403, 'invalid_api_key', 'auth')
-  end
+    it 'fails to get settings if public key used' do
+      new_client = MeiliSearch::Client.new(URL, public_key['key'])
+      expect do
+        new_client.index(random_uid).settings
+      end.to raise_meilisearch_api_error_with(403, 'invalid_api_key', 'auth')
+    end
 
-  it 'fails to get keys if private key used' do
-    private_key = client.keys['private']
-    new_client = MeiliSearch::Client.new(URL, private_key)
-    expect do
-      new_client.keys
-    end.to raise_meilisearch_api_error_with(403, 'invalid_api_key', 'auth')
-  end
+    it 'fails to get keys if private key used' do
+      new_client = MeiliSearch::Client.new(URL, private_key['key'])
+      expect do
+        new_client.keys
+      end.to raise_meilisearch_api_error_with(403, 'invalid_api_key', 'auth')
+    end
 
-  it 'fails to search if no key used' do
-    new_client = MeiliSearch::Client.new(URL)
-    expect do
-      new_client.index(random_uid).settings
-    end.to raise_meilisearch_api_error_with(401, 'missing_authorization_header', 'auth')
-  end
+    it 'fails to search if no key used' do
```

Comment:
What about "fails to get settings if no key is used", or changing the expectation to a search. 🍶

Reply:
Good catch! The title of the test is wrong!